internal/exec/engine.go

```diff
@@ -76,7 +77,7 @@ func (e Engine) Run(stageName string) bool {
 
 	e.Logger.PushPrefix(stageName)
 	defer e.Logger.PopPrefix()
-	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
+	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root, e.OEM).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
```

Comment:
This smells. There is an awful lot of global state flying around at this point. Is there a better way to make this decision?